Sources/Prometheus/MetricTypes/Summary.swift

```diff
@@ -167,6 +168,21 @@ public class PromSummary<NumType: DoubleRepresentable, Labels: SummaryLabels>: P
             self.values.append(value)
         }
     }
+    
+    /// Time the duration of a closure and observe the resulting time in seconds.
+    ///
+    /// - parameters:
+    ///     - labels: Labels to attach to the resulting value.
+    ///     - body: Closure to run & record.
+    @inlinable
+    public func time<T>(_ labels: Labels? = nil, _ body: @escaping () throws -> T) rethrows -> T {
+        let start = DispatchTime.now().uptimeNanoseconds
```

Comment:
I thought NIO had something for this, but looks like `NIODeadline` is [calling `DispatchTime.now()` too!](https://github.com/apple/swift-nio/blob/6174c9802c91868ea4f3daf2fec9cd16be548fd1/Sources/NIO/EventLoop.swift#L393)